src/DataSources/Clients.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+require('regenerator-runtime/runtime');
+const _ = require('lodash');
+
+class ClientDataSource {
+  async initialize(config) {
+    this.clients = {};
+
+    if (await config.get('pay')) {
+      this.clients.PayClient = require('classy-pay-client')({
+        apiUrl: config.get('pay.apiUrl'),
+        timeout: config.get('pay.timeout'),
+        token: config.get('PAY_TOKEN'),
+        secret: config.get('PAY_SECRET')
+      });
+    }
+
+    if (await config.get('api')) {
+      this.clients.ApiClient = require('classy-api-client')({
+        clientId: config.get('APIV2_CLIENT_ID'),
+        clientSecret: config.get('APIV2_CLIENT_SECRET'),
+        timeout: config.get('api.timeout'),
+        oauthUrl: config.get('api.oauthUrl'),
+        apiUrl: config.get('api.apiUrl')
+      });
+    }
+  }
+
+  get(key) {
+    return _.get(this.clients, key, null);
+  }
+}
+
+module.exports = new ClientDataSource;
```

Comment:
If each class will only be instantiated once, you could also use object shorthand notation (unless you're concerned about logging?):
```
module.exports = {
  async initialize() { ... },
  get(key) { ... }
};
```

Reply:
The above will work too (and is how I originally wrote this), but it forfeits the ability to have a constructor which means if I want something to be set before get(key) is called, I need it to be stored in a global variable and initialized before the module.exports statement.

I'd like to move to the class syntax when possible.  I know it's all the same under the hood, but I just find the "declare a class with a constructor and ivars and methods and instantiate it" model so much easier to wrap my head around (maybe because of my background in more strongly typed languages).

Reply:
I don't quite understand what you would pass to a constructor that you wouldn't pass to initialize(). If you did want to pass something to a constructor here in this file, you'd have to require() a config file or something anyway, since the instance is not created by other code but by the module itself. I may just not quite understand the long-term ecosystem in store here?

I do agree that the class syntax is nice; I also used it to create singletons until recently. May just be a matter of preference.